Userland/Libraries/LibGL/SoftwareRasterizer.cpp

```diff
@@ -41,147 +42,135 @@ static Gfx::RGBA32 to_rgba32(const FloatVector4& v)
-        return point.x() >= zero.x()
-            && point.y() >= zero.y()
-            && point.z() >= zero.z();
+    // This function tests whether a point as identified by its 3 edge values lies within the triangle
+    auto test_point = [zero](const IntVector3& edges) -> bool {
+        return edges.x() >= zero.x()
+            && edges.y() >= zero.y()
+            && edges.z() >= zero.z();
     };
 
-    // Calculate bounds
-    FloatVector2 min { AK::min(v0.x, AK::min(v1.x, v2.x)), AK::min(v0.y, AK::min(v1.y, v2.y)) };
-    FloatVector2 max { AK::max(v0.x, AK::max(v1.x, v2.x)), AK::max(v0.y, AK::max(v1.y, v2.y)) };
-
     // Calculate block-based bounds
-    int iminx = floorf(min.x);
-    int iminy = floorf(min.y);
-    int imaxx = ceilf(max.x);
-    int imaxy = ceilf(max.y);
-
-    iminx = clamp(iminx, 0, render_target.width() - 1);
-    imaxx = clamp(imaxx, 0, render_target.width() - 1);
-    iminy = clamp(iminy, 0, render_target.height() - 1);
-    imaxy = clamp(imaxy, 0, render_target.height() - 1);
-
-    int bx0 = iminx / RASTERIZER_BLOCK_SIZE;
-    int bx1 = imaxx / RASTERIZER_BLOCK_SIZE + 1;
-    int by0 = iminy / RASTERIZER_BLOCK_SIZE;
-    int by1 = imaxy / RASTERIZER_BLOCK_SIZE + 1;
+    const int bx0 = max(0, min(min(v0.x, v1.x), v2.x)) / RASTERIZER_BLOCK_SIZE;
+    const int bx1 = min(render_target.width(), max(max(v0.x, v1.x), v2.x) + RASTERIZER_BLOCK_SIZE - 1) / RASTERIZER_BLOCK_SIZE;
+    const int by0 = max(0, min(min(v0.y, v1.y), v2.y)) / RASTERIZER_BLOCK_SIZE;
+    const int by1 = min(render_target.height(), max(max(v0.y, v1.y), v2.y) + RASTERIZER_BLOCK_SIZE - 1) / RASTERIZER_BLOCK_SIZE;
 
     // Iterate over all blocks within the bounds of the triangle
     for (int by = by0; by < by1; by++) {
         for (int bx = bx0; bx < bx1; bx++) {
 
-            // The 4 block corners
-            int x0 = bx * RASTERIZER_BLOCK_SIZE;
-            int y0 = by * RASTERIZER_BLOCK_SIZE;
-            int x1 = bx * RASTERIZER_BLOCK_SIZE + RASTERIZER_BLOCK_SIZE;
-            int y1 = by * RASTERIZER_BLOCK_SIZE + RASTERIZER_BLOCK_SIZE;
-
-            // Barycentric coordinates of the 4 block corners
-            auto a = barycentric_coordinates(x0, y0);
-            auto b = barycentric_coordinates(x1, y0);
-            auto c = barycentric_coordinates(x0, y1);
-            auto d = barycentric_coordinates(x1, y1);
+            // Edge values of the 4 block corners
+            auto b0 = calculate_edge_values({ bx * RASTERIZER_BLOCK_SIZE, by * RASTERIZER_BLOCK_SIZE });
+            auto b1 = calculate_edge_values({ bx * RASTERIZER_BLOCK_SIZE + RASTERIZER_BLOCK_SIZE, by * RASTERIZER_BLOCK_SIZE });
+            auto b2 = calculate_edge_values({ bx * RASTERIZER_BLOCK_SIZE, by * RASTERIZER_BLOCK_SIZE + RASTERIZER_BLOCK_SIZE });
+            auto b3 = calculate_edge_values({ bx * RASTERIZER_BLOCK_SIZE + RASTERIZER_BLOCK_SIZE, by * RASTERIZER_BLOCK_SIZE + RASTERIZER_BLOCK_SIZE });
```

Comment:
`// clang-format off` to keep the nice spacing for the args?
or maybe split them to one arg per line?

(just commenting about this since it was more readable before)

Reply:
`// clang-format off` as a comment?

Reply:
Yes, and to turn it back on, `// clang-format on`